pbft/pbft_failure_detector.cpp

```diff
@@ -98,4 +97,18 @@ pbft_failure_detector::register_failure_handler(std::function<void()> handler)
     std::lock_guard<std::mutex> lock(this->lock);
 
     this->failure_handler = handler;
+}
+
+void
+pbft_failure_detector::add_completed_request_hash(const bzn::hash_t& request_hash)
+{
+    this->completed_requests.emplace(request_hash);
+    this->completed_request_queue.push(request_hash);
+
+    if (max_completed_requests_memory < this->completed_requests.size())
```

Comment:
I prefer while to if here; it shouldn't ever be different but if it is, then removing everything over the limit is the correct behavior. Suppose in the future we change the options module to support modifying options at runtime, for xample.